spec/idempo_spec.rb

```diff
@@ -13,25 +14,23 @@
   describe "with a very large response body" do
     let(:app) do
       the_app = ->(env) {
-        body = Enumerator.new do |yielder|
-          yielder.yield(Random.new.bytes(15))
-          yielder.yield(env["rack.input"].read)
-        end
-        [200, {"X-Foo" => "bar", "Content-Length" => "9999999999999"}, body]
+        pre_sized_body = PreSizedBody.new(999999999)
```

Comment:
**suggestion:** Interesting trick with `PreSizedBody`!

I see that we're not testing small request body doesn't get saved. Should be easy to add with `PreSizedBody` class?

Reply:
A small request body should get saved. What we are not testing is that while the body is not saved, there is still concurrency protection - but testing that would need bypassing rack-spec, because there is no provision for multiple Fibers running simultaneous requests in it.

Reply:
Ah, true! Misread the code, thanks for pointing that out.